[TT-16171] docs: add documentation for default TLS certificate renewal feature (#1326)
TykTechnologies/tyk-docs: snippets/default-tls-certificate.mdx

> *Default TLS Certificate*

Reviewer: ```suggestion
*Gateway TLS Server Certificate*
```

> The Gateway Helm chart ships with a default self-signed TLS certificate for quick evaluation and testing purposes. This certificate is automatically used when you enable TLS via `global.tls.gateway` without providing your own certificate.

Reviewer: ```suggestion
The Gateway Helm chart ships with an example self-signed server certificate for quick evaluation and testing purposes. This certificate is automatically used when you enable TLS via `global.tls.gateway` without providing your own certificate.
```

<Warning>

> The default certificate is self-signed and intended only for development and testing purposes. For production deployments, you should always use your own trusted certificate.

Reviewer: ```suggestion
The example certificate is self-signed and provided as a default, only for development and testing purposes. For production deployments, you should always use your own trusted certificate.
```

</Warning>

> *Transitioning to the New Default Certificate*

Reviewer: ```suggestion
*Transitioning to a New Default Certificate*
```

> Tyk periodically renews the default self-signed certificate shipped with the Helm charts. To ensure backward compatibility during transitions, the chart supports serving both the legacy and the new certificate simultaneously. This allows clients to continue connecting while you update their trust stores.

Reviewer: ```suggestion
Tyk periodically renews the default self-signed certificate shipped with the Helm charts. To ensure continuity during transitions, the chart supports serving both previous and new certificates simultaneously. This allows clients to continue connecting with the previous certificate while you update their trust stores.
```

> To enable the new default certificate alongside the legacy one:

Reviewer: ```suggestion
To enable a new default certificate alongside the existing one:
```

```yaml
tyk-gateway:
  gateway:
    tls:
      useNewDefaultCertificate: true
```

When `useNewDefaultCertificate` is set to `true`:

> - Both the legacy and the new certificates are loaded by the Gateway

Reviewer: ```suggestion
- Both the existing and new certificates are loaded by the Gateway
```
